### PR TITLE
Change Databricks image from 8.2 to 9.1 [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Databricks 8.2 has been deprecated, change to Databricks 9.1 LTS.

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

**Comment by NvTimLiu** [comment]
build

File: jenkins/Jenkinsfile-blossom.premerge
Comment (pxLi) [inline jenkins/Jenkinsfile-blossom.premerge:393]: please also update stage expression,

db_build && major_ver >= 21 && minor_ver >= 6
==>
db_build && major_ver >= 21 && **minor_ver >= 12**

as we added spark312db shim later for 21.12 or later

**Review by pxLi — COMMENTED** [review]
since this change includes the jenkinsfile change, so it would possibly failed the test and we may need to skip ci here, can you test against https://github.com/pxLi/spark-rapids/ repo?

File: jenkins/Jenkinsfile-blossom.premerge
Comment (NvTimLiu) [reply jenkins/Jenkinsfile-blossom.premerge:393]: fixed

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
@tgravescs @jlowe @revans2  Could you please help to review? Thanks!

Please keep this PR `OPEN`, I'll merge on my own it after all the tests/verify pipelines get PASS. Thanks

**Review by revans2 — COMMENTED** [review]
This looks fine to me, but I am not the databricks expert so I will not approve this until someone else with more experience takes a look too.

**Comment by jlowe** [comment]
> Please keep this PR OPEN, I'll merge it after all the tests/verify pipelines get PASS.

Converting this to a draft PR so it cannot be accidentally merged.

**Review by tgravescs — COMMENTED** [review]
this can be done separately but it would be nice to update the jenkins/databricks/build.sh and test.sh script to default to 9.1.

**Review by jlowe — DISMISSED** [review]

**Comment by tgravescs** [comment]
@timl has this been tested?

File: jenkins/databricks/init_cudf_udf.sh
Comment (NvTimLiu) [inline jenkins/databricks/init_cudf_udf.sh:28]: We change to create a common 'cudf-udf' conda environment for cudf udf tests, because 

1, Not like 7.3 & 8.2, Databricks 9.1 does not have the default `databricks-ml-gpu` conda ENV

2, In 7.3 & 8.2, the script init_cudf_udf.sh  create the empty MAMBA conda environment, but install everything (using MAMBA)into the  default `databricks-ml-gpu` ENV, this looks odd.

3, We create a common and clean `cudf-udf` ENV specified for the cudf udf test.

**Comment by NvTimLiu** [comment]
> @NvTimLiu has this been tested?

Yes, this has been tested against both 7.3 & 9.1 build/IT runtime.

On 7.3, got everything PASS, 

On 9.1, FAILED on the issue: https://github.com/rapidsai/cudf/issues/9622

Will verify after the issue get fixed.

File: jenkins/databricks/test.sh
Comment (NvTimLiu) [inline jenkins/databricks/test.sh:26]: Uuse "cudf-udf" conda environment (created by jenkins/databricks/init_cudf_udf.sh) for the python cudf-udf tests.

**Comment by NvTimLiu** [comment]
DB 9.1 tests failed on https://github.com/NVIDIA/spark-rapids/issues/4069

**Review by pxLi — APPROVED** [review]

**Comment by NvTimLiu** [comment]
build

**Comment by pxLi** [comment]
Going to create a dummy verification PR

**Pull request merged**
